lib/charms/prometheus_k8s/v1/prometheus_remote_write.py

```diff
@@ -510,10 +510,25 @@ def reload_alerts(self) -> None:
         """Reload alert rules from disk and push to relation data."""
         self._push_alerts_to_all_relation_databags(None)
 
+    def _deduplicate_endpoints(self, endpoints: List[Dict[str, Any]]) -> List[Dict[str, Any]]:
```

Comment:
I may be wrong, but this is equivalent to `[dict(t) for t in {tuple(d.items()) for d in endpoints}]`

Reply:
Good point, done!